### DISCUSSION TITLE
FIT Scan - What is a Fit Scan for and When do Instrument Scientists Use it?

### DISCUSSION THREAD

**Discussion by JackEAllen** [ISISComputingGroup/IBEX#7359] [Reflectometry] [answered]
I noticed an Instrument scientists running a fit scan using a command similar to: 
`Fit scan command: x=g.cget(b.SMOFFSET)["value"];fit=scan(b.SMOFFSET, x-2, x+2, 15, frames=80,det=280, mon=2,pixel_range=0,fit=Erf) `

Does anyone know what a Fit scan does and is able to describe what is going on exactly in the above command?
I would also like to know how this fits into an instrument scientist workflow if anyone knows?

**Comment by Tom-Willemsen** [comment, marked as answer]
`scan` comes from the [scans library](https://github.com/ISISNeutronMuon/InstrumentScripts/tree/master/general/scans).

Fundamentally it is attempting to plot a graph of detector intensity against `b.SMOFFSET`, and then fit a particular model to this graph.
- It's scanning the value of `b.SMOFFSET` around it's current value (plus or minus 2), counting for 80 frames per measurement.
- The fit function being used is `Erf`, implying that the scientists are looking for some kind of "edge" in their data. Another common fit would be a `Gaussian`, where they want to find the position that will maximise their beam intensity at the detector. They would then use this to optimise the positions on their beamline.
